tests/e2e_logging.py

```diff
@@ -84,6 +84,27 @@ def run(args):
                     )
                     check(c.rpc("LOG_get", {"id": 44}), result=long_msg)
 
+                with primary.user_client(format="json") as c:
```

Comment:
These could be run in a loop instead and re-using the same `"id"` is fine too.

Reply:
It's easy to spot which one failed if you don't re-use the id though

Reply:
Agreed - this could just be in a loop, but separate `id`s keep the tests cleaner.
Something like
```
with primary.user_client(format="json") as c:
  id = 44
  for p in range(14, 20):
    long_msg = "X" * (2 ** p)
    id += 1
    ...
```